jumanji/training/networks/robot_warehouse/actor_critic.py

```diff
@@ -0,0 +1,169 @@
+    MultiCategoricalParametricDistribution,
+)
+from jumanji.training.networks.transformer_block import TransformerBlock
+
+
+def make_actor_critic_networks_robot_warehouse(
+    robot_warehouse: RobotWarehouse,
+    transformer_num_blocks: int,
+    transformer_num_heads: int,
+    transformer_key_size: int,
+    transformer_mlp_units: Sequence[int],
+) -> ActorCriticNetworks:
+    """Make actor-critic networks for the `RobotWarehouse` environment."""
+    num_values = np.asarray(robot_warehouse.action_spec().num_values)
+    parametric_action_distribution = MultiCategoricalParametricDistribution(
+        num_values=num_values
+    )
+    policy_network = make_actor_network(
+        time_limit=robot_warehouse.time_limit,
+        transformer_num_blocks=transformer_num_blocks,
+        transformer_num_heads=transformer_num_heads,
+        transformer_key_size=transformer_key_size,
+        transformer_mlp_units=transformer_mlp_units,
+    )
+    value_network = make_critic_network(
+        time_limit=robot_warehouse.time_limit,
+        transformer_num_blocks=transformer_num_blocks,
+        transformer_num_heads=transformer_num_heads,
+        transformer_key_size=transformer_key_size,
+        transformer_mlp_units=transformer_mlp_units,
+    )
+    return ActorCriticNetworks(
+        policy_network=policy_network,
+        value_network=value_network,
+        parametric_action_distribution=parametric_action_distribution,
+    )
+
+
+class RobotWarehouseTorso(hk.Module):
+    def __init__(
+        self,
+        transformer_num_blocks: int,
+        transformer_num_heads: int,
+        transformer_key_size: int,
+        transformer_mlp_units: Sequence[int],
+        env_time_limit: int,
+        name: Optional[str] = None,
+    ):
+        super().__init__(name=name)
+        self.transformer_num_blocks = transformer_num_blocks
+        self.transformer_num_heads = transformer_num_heads
+        self.transformer_key_size = transformer_key_size
+        self.transformer_mlp_units = transformer_mlp_units
+        self.model_size = transformer_num_heads * transformer_key_size
+        self.embedder = hk.Linear(self.model_size)
+        self.env_time_limit = env_time_limit
+
+    def __call__(self, observation: Observation) -> chex.Array:
+        # Shape names:
+        # B: batch size
+        # N: number of agents
+        # O: observation size
+        # H: hidden/embedding size
+        # (B, N, O)
+        _, num_agents, _ = observation.agents_view.shape
+
+        percent_done = observation.step_count / self.env_time_limit
+        step = jnp.repeat(percent_done[:, None], num_agents, axis=-1)[..., None]
+        agents_view = observation.agents_view
+
+        # join step count and agent view to embed both at the same time
+        # (B, N, O + 1)
+        obs = jnp.concatenate((agents_view, step), axis=-1)
+        # (B, N, O + 1) -> (B, N, H)
+        embeddings = self.embedder(obs)
```

Comment:
Could be defined here directly
```suggestion
        embeddings = hk.Linear(self.model_size)(obs)
```